src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,236 @@
+        }
+
+        if (state.getValue(SOUTH))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.SOUTH);
+        }
+
+        if (state.getValue(WEST))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.WEST);
+        }
+
+        return i;
+    }
+
+    /**
+     * Get the actual Block state of this Block at the given position. This applies properties not visible in the
+     * metadata, such as fence connections.
+     */
+    @NotNull
+    public IBlockState getActualState(@NotNull IBlockState state, @NotNull IBlockAccess worldIn, @NotNull BlockPos pos)
+    {
+        return state.withProperty(NORTH, canPaneConnectTo(worldIn, pos, EnumFacing.NORTH))
+                .withProperty(SOUTH, canPaneConnectTo(worldIn, pos, EnumFacing.SOUTH))
+                .withProperty(WEST, canPaneConnectTo(worldIn, pos, EnumFacing.WEST))
+                .withProperty(EAST, canPaneConnectTo(worldIn, pos, EnumFacing.EAST));
+    }
+
+/**
+ * Used to determine ambient occlusion and culling when rebuilding chunks for render
+ */
+public boolean isOpaqueCube(@NotNull IBlockState state)
+{
+    return false;
+}
+
+    public boolean isFullCube(@NotNull IBlockState state)
+    {
+        return false;
+    }
+
+    private boolean canPaneConnectToBlock(Block blockIn)
+    {
+        return blockIn.getDefaultState().isFullCube() || blockIn == this || blockIn == Blocks.GLASS || blockIn == Blocks.STAINED_GLASS || blockIn == Blocks.STAINED_GLASS_PANE || blockIn instanceof BlockPaperwall;
+    }
+
+    @SideOnly(Side.CLIENT)
+    public boolean shouldSideBeRendered(@NotNull IBlockState blockState, @NotNull IBlockAccess blockAccess, @NotNull BlockPos pos, @NotNull EnumFacing side)
+    {
+        return blockAccess.getBlockState(pos.offset(side)).getBlock() != this && super.shouldSideBeRendered(blockState, blockAccess, pos, side);
+    }
+
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.TRANSLUCENT;
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    public int getMetaFromState(@NotNull IBlockState state)
+    {
+        return 0;
+    }
+
+    /**
+     * Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed
+     * blockstate.
+     */
+    @NotNull
+    public IBlockState withRotation(@NotNull IBlockState state, @NotNull Rotation rot)
+    {
+        switch (rot)
+        {
+            case CLOCKWISE_180:
+                return state.withProperty(NORTH, state.getValue(SOUTH)).withProperty(EAST, state.getValue(WEST)).withProperty(SOUTH, state.getValue(NORTH)).withProperty(WEST, state.getValue(EAST));
+            case COUNTERCLOCKWISE_90:
+                return state.withProperty(NORTH, state.getValue(EAST)).withProperty(EAST, state.getValue(SOUTH)).withProperty(SOUTH, state.getValue(WEST)).withProperty(WEST, state.getValue(NORTH));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 197 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)